src/storage/mvcc/txn.rs

```diff
@@ -128,6 +128,29 @@ impl<'a> MvccTxn<'a> {
         self.unlock_key(key.clone());
         Ok(())
     }
+
+    pub fn gc(&mut self, key: &Key, safe_point: u64) -> Result<()> {
+        let mut after_safe_point = false;
+        let mut ts: u64 = u64::max_value();
+        while let Some((start, commit)) = try!(self.reader.seek_write(key, ts)) {
+            if !after_safe_point {
+                if commit <= safe_point {
+                    // Set `after_safe_point` after the latest write after `safe_point`.
+                    after_safe_point = true;
+                    // Latest write can be deleted if it's a `Modify::Delete`.
+                    if try!(self.reader.load_data(key, start)).is_none() {
```

Comment:
@ngaut What do you mean by `transaction doesn't know the result`? The case is like we send a 'Delete' on a key that does not exist, the server may either save a 'Delete' log or do nothing, because the key does not exist in the first place.